src/Storages/StorageMergeTree.cpp

```diff
@@ -499,7 +513,10 @@ CancellationCode StorageMergeTree::killMutation(const String & mutation_id)
     global_context.getMergeList().cancelPartMutations({}, to_kill->block_number);
     to_kill->removeFile();
     LOG_TRACE(log, "Cancelled part mutations and removed mutation file " << mutation_id);
-    mutation_wait_event.notify_all();
+    {
+        std::unique_lock lock(mutation_wait_mutex);
```

Comment:
why we need it? We use this `cv` just as a notifier, we don't have any critical sections here.

Reply:
Because if mutation is killed before query with `sync_mutations` wait for it to complete (or during running its predicate), the signal will be missed and query will lock until one of next mutations finishes.